### ISSUE TITLE
deleting points after launching the plugin doesn't update plugin

### ISSUE THREAD

**Issue by jni** [brisvag/napari-properties-plotter#2]
If I select points with the plugin and then delete them, the plots don't update. Attempting further selections then causes index errors in napari:

```pytb
IndexError                                Traceback (most recent call last)
~/conda/envs/all/lib/python3.8/site-packages/napari_properties_plotter/property_plotter.py in on_selection_changed(self=<napari_properties_plotter.property_plotter.PropertyPlotter object>, start=0.0, end=31.1220130402246)
    304             x_prop = self.picker.x
    305             idx = np.where(np.logical_and(start <= x_prop, x_prop <= end))[0]
--> 306             layer.selected_data = idx
        layer.selected_data = {269, 270, 278, 281, 156, 286, 288, 295, 296, 300, 303, 305, 306, 308, 309, 182, 310, 311, 313, 315, 316, 317, 319, 320, 321, 322, 323, 324, 326, 327, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 218, 224}
        idx = array([156, 182, 218, 224, 269, 270, 278, 281, 286, 288, 295, 296, 300,
       303, 305, 306, 308, 309, 310, 311, 313, 315, 316, 317, 319, 320,
       321, 322, 323, 324, 326, 327, 330, 331, 332, 333, 334, 335, 336,
       337, 338, 339, 340, 341, 342, 343, 344])

~/projects/napari/napari/layers/points/points.py in selected_data(self=<Points layer 'properties'>, selected_data=array([156, 182, 218, 224, 269, 270, 278, 281, 2...,
       337, 338, 339, 340, 341, 342, 343, 344]))
   1079             return
   1080         index = list(self._selected_data)
-> 1081         edge_colors = np.unique(self.edge_color[index], axis=0)
        edge_colors = undefined
        global np.unique = <function unique at 0x7fc17416a1f0>
        self.edge_color = array([[0., 0., 0., 1.],
       [0., 0., 0., 1.],
       [0., 0., 0., 1.],
       ...,
       [0., 0., 0., 1.],
       [0., 0., 0., 1.],
       [0., 0., 0., 1.]])
        global index = undefined
        global axis = undefined
   1082         if len(edge_colors) == 1:
   1083             edge_color = edge_colors[0]

IndexError: index 300 is out of bounds for axis 0 with size 299
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/conda/envs/all/lib/python3.8/site-packages/napari_properties_plotter/property_plotter.py in on_selection_changed(self=<napari_properties_plotter.property_plotter.PropertyPlotter object>, start=0.0, end=35.337233648084364)
    304             x_prop = self.picker.x
    305             idx = np.where(np.logical_and(start <= x_prop, x_prop <= end))[0]
--> 306             layer.selected_data = idx
        layer.selected_data = {257, 269, 270, 278, 281, 156, 286, 288, 291, 295, 296, 300, 303, 304, 305, 306, 307, 308, 309, 182, 310, 311, 313, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 326, 327, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 216, 344, 218, 224, 245, 248, 249}
        idx = array([156, 182, 216, 218, 224, 245, 248, 249, 257, 269, 270, 278, 281,
       286, 288, 291, 295, 296, 300, 303, 304, 305, 306, 307, 308, 309,
       310, 311, 313, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324,
       326, 327, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340,
       341, 342, 343, 344])

~/projects/napari/napari/layers/points/points.py in selected_data(self=<Points layer 'properties'>, selected_data=array([156, 182, 216, 218, 224, 245, 248, 249, 2..., 337, 338, 339, 340,
       341, 342, 343, 344]))
   1079             return
   1080         index = list(self._selected_data)
-> 1081         edge_colors = np.unique(self.edge_color[index], axis=0)
        edge_colors = undefined
        global np.unique = <function unique at 0x7fc17416a1f0>
        self.edge_color = array([[0., 0., 0., 1.],
       [0., 0., 0., 1.],
       [0., 0., 0., 1.],
       ...,
       [0., 0., 0., 1.],
       [0., 0., 0., 1.],
       [0., 0., 0., 1.]])
        global index = undefined
        global axis = undefined
   1082         if len(edge_colors) == 1:
   1083             edge_color = edge_colors[0]

IndexError: index 300 is out of bounds for axis 0 with size 299
````

The [`remove_points`](https://github.com/napari/napari/blob/ed6b8ef2e55471b42a3ea47875b7e1ef0f99ba80/napari/layers/points/points.py#L1483-L1501) method uses the `self.data = ` setter and so emits a `.data` event, so the plot/plugin should listen to those events and update itself accordingly.

**Comment by jni** [comment]
reproducer:

```python
import napari
import numpy as np


viewer = napari.Viewer()

coords = np.random.random((20, 2)) * [512, 512]
props = dict(zip('abc', np.random.random((3, 20))))
viewer.add_points(coords, properties=props)

napari.run()
```

Then launch points properties plotter, plot 'a' vs 'b', select area, switch back to napari and change the points mode to "select", then press backspace. Finally, adjust the selected area in the properties plotter.

**Comment by brisvag** [comment]
Fixed by #6 .
